src/NServiceBus.Core/Unicast/Transport/TransportConnectionString.cs

```diff
@@ -33,21 +47,15 @@ public string GetConnectionStringOrRaiseError(TransportDefinition transportDefin
             return connectionString;
         }
 
-        static string ReadConnectionString(string connectionStringName)
-        {
-            var connectionStringSettings = ConfigurationManager.ConnectionStrings[connectionStringName];
-
-            return connectionStringSettings?.ConnectionString;
-        }
-
-        Func<string> GetValue = () => ReadConnectionString(DefaultConnectionStringName);
+        Func<string> GetValue;
 
+#if NET452
```

Comment:
Maybe I'm not following exactly, but having one class for `TransportConnectionString452` and another for `TransportConnectionStringNetStandard` would only serve to push the `#if` logic on level higher in the code. So instead of being contained within in this one class/file, we'd have `#if` logic in about 11 other places (according to find usages). 

There would also be duplicate code between the two classes so do we abstract that away by creating a 3rd base class? 

Honestly, any of the solutions that I see for moving the `#if` logic out of this class end up in a leaky abstraction....or am I not seeing the solution you are?